roles/caw/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+
+- name: Fetch CAW from GitHub
+  git: repo="{{ caw_repo }}"
+       dest="{{ caw_dest }}"
+       version="{{ caw_version }}"
+       force=yes
+
+- name: Create CAW Singularity folder
+  file: name="{{ caw_singularity }}"
+        state=directory mode=g+s
+
+- name: Create temp directory for CAW singularity images
+  command: "mkdir {{ item }}"
+  args:
+    creates: "{{ item }}"
+  with_items:
+  - "{{ caw_singularity }}/singularity_cache"
+  - "{{ caw_singularity }}/singularity_temp"
+
+- name: Pull CAW Singularity images
+  command: "singularity build {{ item }}-{{ caw_version }}.img {{ caw_singularity_repo }}/{{ item }}:{{ caw_version }}"
+  environment:
+    SINGULARITY_CACHEDIR: "{{ caw_singularity }}/singularity_cache"
+    SINGULARITY_TMPDIR: "{{ caw_singularity }}/singularity_temp"
+  args:
```

Comment:
Perhaps a `creates` can be added here if you know how the output of the path for the build will look like? Or - if we do not care about keeping several versions of each container, just point look for the directory path that is used instead?

Reply:
Actually it should not exists before as the `{{ caw_singularity }}` is created each time for new deployment

Reply:
But yes, you are right. Adding `creates` is not gonna hurt and also might be useful while test deploying

Reply:
Ah, yeah, that will be true in most of the cases, indeed. But doesn't hurt as you say, as it will speed things up e.g re-building, as well making it a bit more future proof.